docs/performance/productizing.rst

```diff
@@ -0,0 +1,116 @@
+performance, without impacting other workloads.
+
+Please take a look at :ref:`Query Optimization 101 <performance-optimization>` for strategies to optimize your
+queries when you encounter this situation.
+
+.. _reading-lots-of-records:
+
+*************************
+ Reading lots of records
+*************************
+
+When the http endpoint is used CrateDB will prepare the entire response in
+memory before sending it to the client.
+
+When the PostgreSQL protocol is used CrateDB attempts to stream the results but
+in many cases it still needs to bring all rows to the query handler node first.
+
+So we should always limit how many rows we request at a time, see `Fetching
+large result sets from CrateDB`_
+
+.. _number-of=shards:
+
+******************
+ Number of shards
+******************
+
+In CrateDB data in tables and partitions is distributed in storage units that we
+call shards.
+
+If we do not specify how many shards we want for a table/partition CrateDB will
+derive a default from the number of CPUs.
+
+CrateDB also has replicas of data and this results in additional shards in the
+cluster.
+
+Having too many or too few shards has performance implications, so it is very
+important to get familiar with the :ref:`Sharding Performance Guide <sharding_guide>`
+
+But in particular, there is a soft limit of 1000 shards per node; so table
+schemas, partitioning strategy, and number of nodes need to be planned to stay
+well below this limit, one strategy can be to aim for a configuration where even
+if one node in the cluster is lost the remaining nodes would still have less
+than 1000 shards.
+
+If this was not considered when initially defining the tables we have the
+following considerations:
+
+-  changing the partitioning strategy requires creating a new table and copying
+   over the data
+-  the easiest way to change the number of shards on a partitioned table is to
+   do it for new shards only with the `ALTER TABLE ONLY` command
+-  see also `Changing the number of shards`_
+
+.. _amount-of-indexed-columns:
+
+*****************************
+ Amount of (indexed) columns
+*****************************
+
+CrateDB indexes everything by default, providing excellent query performance
+without requiring manual indexing. However, excessive indexing can impact
+storage, write speed, and resource utilization.
+
+-  All columns are automatically indexed when inserted.
+-  CrateDB optimizes indexing using Lucene-based columnar storage.
+-  A soft limit of 1,000 indexed columns (or JSON fields) per table exists.
+-  Going beyond this limit may impact performance.
+
+In cases with many fields and columns, it is advised to determine which columns
+need to be indexed.
+
+.. _section-joins:
+
+*******
+ JOINs
+*******
+
+CrateDB is a lot better at JOINs than many of our competitors and is getting
+better at every release, but JOINs in distributed databases are tricky to
+optimize, so in many cases queries involving JOINs may need a bit of tweaking.
```

Comment:
I get the message, and I love it. However, I think the sound is a bit sloppy, specifically the bragging fragment

> is a lot better at JOINs than many of our competitors

even if it's true. Maybe we can find a better wording? /cc @kneth, @simonprickett

Reply:
Maybe be more specific, possibly even to the extent of doing name dropping, like...

> CrateDB's engine to coordinate and execute distributed JOINs is already more efficient
> than many competing implementations {by other vendors | like ArangoDB, Cassandra, CockroachDB, MongoDB}, building upon years of maturing since its introduction with CrateDB 1.0.
> Because JOIN operations in distributed databases are tricky to optimize, 
> they need a bit of query tweaking, based on what the query planner says.

So, I've added another variant of bragging about CrateDB's implementation ("building upon years of maturing since its introduction with CrateDB 1.0"), and also another variant of sloppyness ("based on what the query planner says"). I am sure you will make the best out of it?